src/main/java/io/stargate/sgv2/jsonapi/service/operation/reranking/RerankingTask.java

```diff
@@ -149,26 +149,33 @@ float[] sortVector() {
     return sortVector;
   }
 
+  public record DeferredCommandWithSource(
+      Rank.RankSource rankSource, DeferredCommandResult deferredRead) {}
+
   private record DeduplicationResult(
       int totalReads,
       int totalDocuments,
       int droppedDocuments,
       List<ScoredDocument> deduplicatedDocuments) {}
 
-  private DeduplicationResult deduplicateResults(List<CommandResult> rawReadResults) {
+  private DeduplicationResult deduplicateResults() {
 
-    if (rawReadResults.isEmpty()) {
-      throw new IllegalArgumentException("rawReadResults must not be empty");
+    if (deferredReads.isEmpty()) {
+      throw new IllegalArgumentException("deferredReads must not be empty");
```

Comment:
nit: since deferredReads is not an argument, better choice might be `IllegalStateException`.

Reply:
changed